### PR TITLE
Phoebe - Ampers

### PR DESCRIPTION
**Pull request by pmanni02**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | One design decision was whether or not I needed a Room class. After creating a Room class during Wave 1, I realized that I probably would not need a Room class. However, I kept it in case I might need it. This decision worked out until I started working on the cli. Making/starting the cli revealed issues with my code. One major issue (which required lots of late changes) were the instance variables of the Room clas. If I could start over, I would try having Reservations within a Room instance instead of the other way around.
Describe a concept that you gained more clarity on as you worked on this assignment. | Overall design. Design diagrams helped me anticipate many potential issues. Additionally, with the exception of the changes mentioned in the question above, my design process allowed for flexibility with changes.
Describe a _nominal test_ that you wrote for this assignment. | One nominal test was making sure a block hash could be created.
Describe an _edge case test_ that you wrote for this assignment. |  One edge case test was passing an invalid number of rooms for a block into my method for making blocks.
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | I feel I did pretty good. The pseduocode really helped with planning where to start. I could have done more pseudocode near the end of wave 3. It may have helped me avoid the major changes needed at the last minute.

### CODE REVIEW

File: lib/admin.rb [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    def get_unreserved_rooms(date_range)
+      unreserved_room_ids = check_reservations(date_range, reservations)
+
+      desired_start_date = date_range[:start_date]
+      desired_end_date = date_range[:end_date]
+
+      blocks.each do |block|
+        block_start = block[:start_date]
+        block_end = block[:end_date]
+        room_objs = block[:rooms]
```
Comment (kariabancroft) [inline lib/admin.rb:150]: While these variables look nicer, its not actually better to create these local variables that you're only using once rather than using the original variables. You should only use local variables if you'll be using them for a  larger purpose.

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | Yes. You did a nice job separating things out into their separate classes. Also, nice use of private methods.
Demonstrated loose coupling | Yes
Methods demonstrate a good use of encapsulation, inputs and outputs | Yes-looks good
Wave 1 requirements |Yes
Wave 2 requirements |Yes
Wave 3 requirements |Yes
